core/src/raw/http_util/client.rs

```diff
@@ -161,12 +161,10 @@ impl HttpFetch for reqwest::Client {
 
         let (parts, body) = req.into_parts();
 
-        let mut req_builder = self
-            .request(
-                parts.method,
-                reqwest::Url::from_str(&uri.to_string()).expect("input request url must be valid"),
-            )
-            .headers(parts.headers);
+        let url = reqwest::Url::from_str(&uri.to_string())
```

Comment:
Hi, it's better to return an error instead. We can use something like:

```rust
let url = reqwest::Url::from_str(&uri.to_string()).map_err(|err| Error::new("request ur is invalid").context("uri", uri).source(err))?;
```

Reply:
I noticed there is a very similar piece of code at https://github.com/apache/opendal/blob/v0.53.1/core/src/raw/http_util/client.rs#L189-L195

```rust
let mut resp = req_builder.send().await.map_err(|err| {
    Error::new(ErrorKind::Unexpected, "send http request")
        .with_operation("http_util::Client::send")
        .with_context("url", uri.to_string())
        .with_temporary(is_temporary_error(&err))
        .set_source(err)
})?;
```
I am imitating this fragment but omitting the `with_temporary()`.
Should they both be using `with_operation("http_util::Client::fetch")`? (instead of `send`)